### PR TITLE
topic/xml policy

### PR DESCRIPTION
**Pull request by sguiriec**
@plbossart, @cc6565 please review the swith to XML configuration

### CODE REVIEW

File: common/audio/AndroidBoard.mk [hunk 1]
```diff
@@ -3,7 +3,8 @@ LOCAL_PATH := $(call my-dir)
 # audio hardware is not discoverable, select hardware or use basic default
 AUDIO_HARDWARE := default
 #AUDIO_HARDWARE := PCH-CX20724
-#AUDIO_HARDWARE := nuc
+# Next configuration is used for Intel NUC6i5SYH
```
Comment (kalyankondapally) [inline common/audio/AndroidBoard.mk]: Can we have Jira: and Tests as part of commit message. We can create one top level bug here for all Audio changes and close it once work is done https://01.org/jira/projects/AIA/issues

File: common/audio/PCH-CX20724/policy/audio_policy_configuration.xml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                -"sink": the sink involved in this route
+                -"sources": all the sources than can be connected to the sink via vis route
+        “attachedDevices”: permanently attached devices.
+        The attachedDevices section is a list of devices names. The names correspond to device names
+        defined in <devicePorts> section.
+        “defaultOutputDevice”: device to be used by default when no policy rule applies
+    -->
+    <modules>
+        <!-- Primary Audio HAL -->
+        <module name="primary" halVersion="3.0">
```
Comment (plbossart) [inline common/audio/PCH-CX20724/policy/audio_policy_configuration.xml]: 2.0?
Or do we make all HALs 3.0 (baseline and PFW)?

File: common/audio/PCH-CX20724/policy/audio_policy_configuration.xml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                <devicePort tagName="Wired Headphones" type="AUDIO_DEVICE_OUT_WIRED_HEADPHONE" role="sink">
+                    <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                             samplingRates="48000" channelMasks="AUDIO_CHANNEL_OUT_STEREO"/>
+                </devicePort>
+
+                <devicePort tagName="Built-In Mic" type="AUDIO_DEVICE_IN_BUILTIN_MIC" role="source">
+                    <profile name="" format="AUDIO_FORMAT_PCM_16_BIT"
+                             samplingRates="8000,11025,12000,16000,22050,24000,32000,44100,48000"
+                             channelMasks="AUDIO_CHANNEL_IN_MONO,AUDIO_CHANNEL_IN_STEREO,AUDIO_CHANNEL_IN_FRONT_BACK"/>
+                </devicePort>
```
Comment (plbossart) [inline common/audio/PCH-CX20724/policy/audio_policy_configuration.xml]: why do we need _IN_FRONT_BACK?

**Comment by plbossart** [comment]
An additional comment is that you define a PRIMARY tag and also include the stub HAL in the audio_policy_configuration.xml for the NUC. Is the intent that the stub HAL is used when nothing is connected to the device (HDMI or headset)?

In that case, maybe we should also add HDMI in all configuration files, even if there is no connector on the device, and rely on the WiredAccessoryManager to make it active.

File: common/audio/PCH-CX20724/policy/audio_policy_configuration.xml (hunk 2 above)
Comment (sguiriec) [inline common/audio/PCH-CX20724/policy/audio_policy_configuration.xml]: We should swith to 3.0 for Baseline and Parameter Framework base.

File: common/audio/PCH-CX20724/policy/audio_policy_configuration.xml (hunk 3 above)
Comment (sguiriec) [inline common/audio/PCH-CX20724/policy/audio_policy_configuration.xml]: We can remove it.

**Comment by plbossart** [comment]
I thought I had merge rights for audio stuff but apparently not. 
@kalyankondapally, can you handle this? Thanks!

**Pull request merged**